src/test/java/bio/terra/service/filedata/azure/tables/TableDirectoryDaoConnectedTest.java

```diff
@@ -86,23 +84,53 @@ public void cleanup() throws Exception {
     // Should already be deleted
     for (String entry : directoryEntriesToCleanup) {
       try {
-        tableDirectoryDao.deleteDirectoryEntry(tableServiceClient, entry);
+        tableDirectoryDao.deleteDirectoryEntry(
+            tableServiceClient, datasetId, DATASET_TABLE.toTableName(), entry);
       } catch (Exception ex) {
         logger.debug("Directory entry either already deleted or unable to delete {}", entry, ex);
       }
     }
-    if (tableName != null) {
+    if (snapshotTableName != null) {
       try {
-        TableClient tableClient = tableServiceClient.getTableClient(tableName);
+        TableClient tableClient = tableServiceClient.getTableClient(snapshotTableName);
         tableClient.deleteTable();
       } catch (Exception ex) {
-        logger.error("Unable to delete table {}", tableName, ex);
+        logger.error("Unable to delete table {}", snapshotTableName, ex);
       }
     }
 
     connectedOperations.teardown();
   }
 
+  @Test
+  public void testStoreTopDirectory() {
+    tableDirectoryDao.storeTopDirectory(tableServiceClient, snapshotId, dataset.getName());
+
+    snapshotTableName = SNAPSHOT_TABLE.toTableName(snapshotId);
+    int count = getTableEntryCount(tableServiceClient, snapshotTableName);
+    assertThat("Store top directory should add two entries to snapshot table.", count, equalTo(2));
+
+    // get directories to confirm the correct ones are added
+    List<String> directories = new ArrayList();
+    directories.add("/");
+    directories.add("/" + dataset.getName());
+
+    List<FireStoreDirectoryEntry> datasetDirectoryEntries =
+        tableDirectoryDao.batchRetrieveByPath(
+            tableServiceClient, snapshotId, snapshotTableName, directories);
+    datasetDirectoryEntries.forEach(d -> directoryEntriesToCleanup.add(d.getFileId()));
+    assertThat("Retrieved entries for all paths", datasetDirectoryEntries.size(), equalTo(2));
+
+    // Test that batchRetrieveByPath only returns unique entries
+    List<String> nonUniqueDirectories = new ArrayList();
+    nonUniqueDirectories.add("/" + dataset.getName());
+    nonUniqueDirectories.add("/_dr_/" + dataset.getName());
+    List<FireStoreDirectoryEntry> uniqueDatasetDirectoryEntries =
+        tableDirectoryDao.batchRetrieveByPath(
+            tableServiceClient, snapshotId, snapshotTableName, nonUniqueDirectories);
+    assertThat("Only 1 unique path", uniqueDatasetDirectoryEntries.size(), equalTo(1));
```

Comment:
Using hamcrest `hasSize()` is a little clearer IMO
```suggestion
    assertThat("Only 1 unique path", uniqueDatasetDirectoryEntries, hasSize(1));
```